tensorflow_probability/bayesian_neural_network/trainer/bayesian_neural_network.py

```diff
@@ -0,0 +1,344 @@
+# Copyright 2018 The TensorFlow Probability Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ============================================================================
+"""Trains a Bayesian neural network to classify MNIST digits.
+
+The architecture is LeNet-5 [1].
+
+#### References
+
+[1]: Yann LeCun, Leon Bottou, Yoshua Bengio, and Patrick Haffner.
+     Gradient-based learning applied to document recognition.
+     _Proceedings of the IEEE_, 1998.
+     http://yann.lecun.com/exdb/publis/pdf/lecun-01a.pdf
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from trainer.tfgfile_wrapper import tfgfile_wrapper
+import os
+import warnings
+
+# Dependency imports
+from absl import flags
+import matplotlib
+matplotlib.use("Agg")
+from matplotlib import figure  # pylint: disable=g-import-not-at-top
+from matplotlib.backends import backend_agg
+import numpy as np
+import tensorflow as tf
+import tensorflow_probability as tfp
+
+from tensorflow.contrib.learn.python.learn.datasets import mnist
+
+# TODO(b/78137893): Integration tests currently fail with seaborn imports.
```

Comment:
Should we remove this TODO?

Reply:
TF is moving out of contrib, what about: tensorflow.keras.datasets